src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -58,11 +60,50 @@ public string EngineName
                 if (engine != value)
                 {
                     engine = value;
+                    SetEnginePackageInfo(engine);
                     RaisePropertyChanged(nameof(EngineName));
                 }
             }
         }
 
+        [JsonProperty("EnginePackageName")]
```

Comment:
Is there value in serializing this with each node?

Reply:
Same graph containing multiple python nodes, pointing to different engines, imported from different packages.